src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -246,7 +293,19 @@ func (mod *mgmtModule) getFabricInterface(ctx context.Context, numaNode int, net
 
 	mod.fabricInfo.CacheScan(ctx, result)
 
-	return mod.fabricInfo.GetDevice(numaNode, netDevClass, provider)
+	return mod.getCachedInterface(ctx, params)
+}
+
+func (mod *mgmtModule) getCachedInterface(ctx context.Context, params *fiParams) (*FabricInterface, error) {
+	if params.interfaceName != "" {
+		fi, err := mod.fabricInfo.localNUMAFabric.FindDevice(params.interfaceName,
+			params.interfaceDomain, params.provider)
+		if err != nil {
+			return nil, err
+		}
+		return fi[0], nil
```

Comment:
Will it always be correct to return the first one? Not fully following the logic here; maybe the thought is that any of them would be fine, so choosing the first is simplest?

Reply:
That's the thought, yeah. If the interface, domain, and provider match, it's sufficient. In practice we'll probably only get one for the combination of all three params, but I wanted this function to be flexible in the other area where we use it (line 198).